File: go/009_palindrome_number.go
package main

import (
	"fmt"
	"strconv"
)

func isPalindrome(x int) bool {
	strX := strconv.Itoa(x)

	for i := 0; i < len(strX)/2; i++ {
		if strX[i] != strX[len(strX)-1-i] {
			return false
		}
	}
	return true
}

func main() {
	fmt.Println(isPalindrome(121))
	fmt.Println(isPalindrome(-121))
	fmt.Println(isPalindrome(10))
}

File: javascript/009_palindromeNumber.js
function isPalindrome(x) {
  var strX = x.toString();
  var end = strX.length - 1;
  for (var z = 0; z < strX.length; z++) {
    if (strX[z] !== strX[end - z]) {
      return false;
    }
  }
  return true;
}
console.log(isPalindrome(121));
console.log(isPalindrome(-121));
console.log(isPalindrome(10));
